lib/web3/requestmanager.js

```diff
@@ -226,9 +226,16 @@ RequestManager.prototype.poll = function () {
         }
 
         results.map(function (result, index) {
-            result.callback = self.polls[index].callback;
-            return result;
+            // make sure the filter is still installed after arrival of the request
+            if(self.polls[index]) {
+                result.callback = self.polls[index].callback;
+                return result;
+            } else
+                return false;
         }).filter(function (result) {
+            if(!result)
+                return false;
```

Comment:
also, it's usually better and cleaner to make a separate filter for this kind of statements